### ISSUE TITLE
[FEATURE] Declarative property to prevent animation from triggering

### ISSUE THREAD

**Issue by this-gavagai** [framer/motion#1201]
**Is your feature request related to a problem? Please describe.**
Often, it is useful to delay an animation until some kind of condition is met. For example, a slider might have text whose animation should not be started until the slide becomes active.

**Describe the solution you'd like**
I'd like to have a boolean property on motion components that prevents animations from triggering if set. This would be similar in function to the `immediate` property in react-spring. When unset, the animation would proceed as normal.

**Describe alternatives you've considered**
The effect I am looking for is currently possible with either imperative functions or variables in the animate property (i.e., `animate={{ x: isReady ? 100 : 0 }}`. Both solutions are acceptable, but they can be unnecessarily complex in situations where an animation is supposed to trigger only once. For example, it could be useful to have an animation trigger when a certain condition is met but not to revert if values change in the future.

Edit: I forgot to mention that one way to do this would be with the delay value (i.e., `transition={{ delay: isReady ? 0 : Infinity }}` or something similar), but it does not appear to be possible to set the delay dynamically like this after the component has mounted.

**Comment by mattgperry** [comment]
Closing this as `wontfix` to set expectations that it probably won't be implemented given it is, while not graceful, possible in userland.
